frontend/packages/ux-editor/src/hooks/queries/useComponentSchemaQuery.ts

```diff
@@ -0,0 +1,33 @@
+// import { useServicesContext } from 'app-shared/contexts/ServicesContext';
+import { useQuery, useQueryClient, UseQueryResult } from '@tanstack/react-query';
+import { addSchemas, dereferenceSchema } from '../../utils/formValidationUtils';
+import { componentSchemaMocks } from '../../testing/componentSchemaMocks';
+
+export interface UseComponentSchemaQueryResult {
+  queryResults: {
+    [key: string]: UseQueryResult<any>;
+  };
+}
+
+// Currently use local mocks rather than fetching from CDN. This is because the CDN schemas are not ready to use, and
+// we also have made some modifications locally to the schemas.
+// When the schemas are available on CDN, we can remove the mocks and use the querys instead.
+export const useComponentSchemaQuery = (component: string): UseQueryResult<any> => {
+  const queryClient = useQueryClient();
+
+  return useQuery(
+    [component],
+    () => {
+      addSchemas([componentSchemaMocks[component]]);
+      return Promise.resolve(dereferenceSchema(componentSchemaMocks[component]));
```

Comment:
Er dette riktig? `componentSchemaMocks` er vel testdata?

Reply:
Ja - mulig vi kan flytte de til et annet sted da for å unngå forvirring. Men se kommentaren på `useComponentSchemaQuery`-metoden for hvorfor det er gjort på denne måten.

Reply:
Jeg så det - trodde jeg hadde slettet denne kommentaren 😅